python/lsst/cp/pipe/ptc/cpPtcSolve.py

```diff
@@ -1077,14 +1110,168 @@ def _makeZeroSafe(self, array, substituteValue=1e-9):
 
         return array
 
+    def fitPtcRolloff(self, dataset):
+        """Fit the photon transfer curve to the
+        Astier+19 approximation (Eq. 16) with a roll-off model to try
+        and capture saturation effects.
+
+        Parameters
+        ----------
+        dataset : `lsst.ip.isr.ptcDataset.PhotonTransferCurveDataset`
+            The dataset containing the means, variances and
+            exposure times.
+
+        Returns
+        -------
+        dataset : `lsst.ip.isr.ptcDataset.PhotonTransferCurveDataset`
+            This is the same dataset as the input parameter, however,
+            it has been modified to include information such as the
+            fit vectors and the fit parameters. See the class
+            `PhotonTransferCurveDatase`.
+        """
+        # For FULLCOVARIANCE model fit
```

Comment:
I don't understand why this function should be erasing the dataset.  The docstring says it adds fields, not wipes them.

Reply:
Maybe it did in an older version?  Either way, this is now modifying in place, and that should be what it does rather than also returning the dataset.